### PR TITLE
Rill Developer UX for partner-filtered dashboards

### PR DESCRIPTION
**Pull request by ericpgreen2**
This PR adds a "View as" button to dashboards in Rill Developer. With this button, a developer can simulate how a user will see the dashboard under the dashboard's granular access policies.

The "View as" button retrieves mock users defined in the `rill.yaml` file under the `mock_users` key. This PR changes the default `rill.yaml` file to include some example mock users.

Here are some example projects that use granular access policies: [Parag's pipemetrics demo](https://github.com/rilldata/pipemetrics-demo), [Eric's margins demo](https://github.com/ericpgreen2/rill-demo-partner-filtered-dashboards) 

Future work:
- Retain the mock user selection when navigating pages (the current behavior is a hack to workaround a bug)
- Add a widget to the `rill.yaml` editor to easily add mock users. Then we can remove the seed users, which will clean up the default `rill.yaml` file.
- Add more type checking when validating access policies

### CODE REVIEW

**Review by djbarnwal — COMMENTED** [review]
Looks good codewise. Have left a small nit comment.

Apart from that, while navigating between users, sometimes, the time controls are hidden in the dashboard.

https://github.com/rilldata/rill/assets/4402679/3dfe0205-50e9-46b8-82c5-f23043076dc7

**Comment by ericpgreen2** [comment]
> Apart from that, while navigating between users, sometimes, the time controls are hidden in the dashboard.

Hmm. I haven't been able to reproduce this myself. It looks like it'd happen if the `/time-range-summary` query doesn't complete. I've tried throttling my network connection but it always resolves for me, just delayed. Any tips on how to reproduce? When you see it, do you see any errors in the browser console?

**Comment by djbarnwal** [comment]
> > Apart from that, while navigating between users, sometimes, the time controls are hidden in the dashboard.
> 
> Hmm. I haven't been able to reproduce this myself. It looks like it'd happen if the `/time-range-summary` query doesn't complete. I've tried throttling my network connection but it always resolves for me, just delayed. Any tips on how to reproduce? When you see it, do you see any errors in the browser console?

Don't see any errors or warnings in the console. But I see that `allTimeRange?.start` in `TimeControls` is undefined when this happens.

**Comment by ericpgreen2** [comment]
> > > Apart from that, while navigating between users, sometimes, the time controls are hidden in the dashboard.
> > 
> > 
> > Hmm. I haven't been able to reproduce this myself. It looks like it'd happen if the `/time-range-summary` query doesn't complete. I've tried throttling my network connection but it always resolves for me, just delayed. Any tips on how to reproduce? When you see it, do you see any errors in the browser console?
> 
> Don't see any errors or warnings in the console. But I see that `allTimeRange?.start` in `TimeControls` is undefined when this happens.

Hmm. I feel I'm debugging blind because I can't reproduce this. It certainly feels related to [this code](https://github.com/rilldata/rill/pull/2965/files#diff-2c5e0cd180c933da53c5b7391681ab51ca6e86218984262a69f03a6bc9960acaR143-R152). Rather than `removeQueries`, we could try `resetQueries`. `invalidateQueries` and `refetchQueries` aren't working for me. Could you try `resetQueries` and see if it fixes it for you? @AdityaHegde, are you able to reproduce this? What do you think?

**Review by nishantmonu51 — APPROVED** [review]

**Comment by nishantmonu51** [comment]
Merging this PR to unblock the release. 
@AdityaHegde : please create a follow up PR for the race condition if you are able to reproduce this.

File: docs/docs/develop/security.md [hunk 1]
```diff
@@ -68,7 +68,7 @@ mock_users:
 - email: anon@unknown.com
 ```
 
-On the dashboard page, provided you've added a security, you'll see a "View as" button in the top right corner. Click this button and select one of your mock users. You'll see the dashboard as that user would see it.
+On the dashboard page, provided you've added a policy, you'll see a "View as" button in the top right corner. Click this button and select one of your mock users. You'll see the dashboard as that user would see it.
```
Comment (pjain1) [inline docs/docs/develop/security.md:71]:
```suggestion
On the dashboard page, provided you've added a security policy, you'll see a "View as" button in the top right corner. Click this button and select one of your mock users. You'll see the dashboard as that user would see it.
```

File: runtime/compilers/rillv1/init.go [hunk 2]
```diff
@@ -16,7 +16,9 @@ func IsInit(ctx context.Context, repo drivers.RepoStore, instanceID string) bool
 
 // InitEmpty initializes an empty project
 func InitEmpty(ctx context.Context, repo drivers.RepoStore, instanceID, title string) error {
-	rillYAML := fmt.Sprintf("compiler: %s\n\ntitle: %q\n", Version, title)
+	mockUsersInfo := "# These are example mock users to test your granular access policies.\n# For more information, see the documentation: https://docs.rilldata.com/develop/security"
```
Comment (pjain1) [inline runtime/compilers/rillv1/init.go:19]:
```suggestion
	mockUsersInfo := "# These are example mock users to test your security policy.\n# For more information, see the documentation: https://docs.rilldata.com/develop/security"
```

File: runtime/compilers/rillv1beta/project.go [hunk 3]
```diff
@@ -31,7 +31,9 @@ func (c *Codec) IsInit(ctx context.Context) bool {
 }
 
 func (c *Codec) InitEmpty(ctx context.Context, title string) error {
-	err := c.Repo.Put(ctx, "rill.yaml", strings.NewReader(fmt.Sprintf("compiler: %s\n\ntitle: %q\n", Version, title)))
+	mockUsersInfo := "# These are example mock users to test your granular access policies.\n# For more information, see the documentation: https://docs.rilldata.com/develop/security"
```
Comment (pjain1) [inline runtime/compilers/rillv1beta/project.go:34]:
```suggestion
	mockUsersInfo := "# These are example mock users to test your security policy.\n# For more information, see the documentation: https://docs.rilldata.com/develop/security"
```

File: web-common/src/runtime-client/RuntimeProvider.svelte [hunk 4]
```diff
@@ -10,6 +12,13 @@
     instanceId: instanceId,
     jwt: jwt,
   });
+
+  const queryClient = useQueryClient();
+
+  // Whenever the runtime's (dev) JWT changes (even to `null`), invalidate all metrics views.
+  // Metrics views may have access policies, for which a JWT grants permission.
```
Comment (pjain1) [inline web-common/src/runtime-client/RuntimeProvider.svelte:19]:
```suggestion
  // Metrics views may have security policy, for which a JWT grants permission.
```

**Pull request merged**